### DISCUSSION TITLE
How to tell if a SWF is Actionscript3 or not?

### DISCUSSION THREAD

**Discussion by catharsis71** [ruffle-rs/ruffle#4248] [Q&A]
I decompiled a SWF that doesn't work properly in Ruffle to try to figure out if it's Actionscript3 or not.  So now I have over 300 files but I'm not really sure what I'm looking for.

Here's a random bit of code, is this enough to tell?

```
   if(ParamsValue != null)
   {
      var _loc3_ = new XML(ParamsValue);
      _loc3_.ignoreWhite = true;
      var _loc2_ = 0;
      while(_loc2_ < _loc3_.childNodes[0].childNodes.length)
      {
         var _loc1_ = _loc3_.childNodes[0].childNodes[_loc2_];
         GlobalPlayerOptions[_loc1_.nodeName] = _loc1_.childNodes[0].nodeValue;
         _loc2_ = _loc2_ + 1;
      }

```

**Comment by Bale001** [comment]
If Ruffle gives you a warning saying the content is AS3, it is AS3. If you saw no warning, it is AS2
